Validation/TrackerRecHits/src/SiStripRecHitsValid.cc

```diff
@@ -624,40 +616,47 @@ void SiStripRecHitsValid::createMEs(DQMStore::IBooker & ibooker,const edm::Event
       createLayerMEs(ibooker,label);
     }
     // book sub-detector plots 
-    auto sdet_pair = folder_organizer.getSubDetFolderAndTag(detid, tTopo);
-    // std::cout << "sdet_pair " << sdet_pair.first << " " << sdet_pair.second << std::endl;
     if (SubDetMEsMap.find(det_layer_pair.first) == SubDetMEsMap.end()){
+      auto sdet_pair = folder_organizer.getSubDetFolderAndTag(detid, tTopo);
+      // std::cout << "sdet_pair " << sdet_pair.first << " " << sdet_pair.second << std::endl;
       ibooker.setCurrentFolder(sdet_pair.first);
       createSubDetMEs(ibooker,det_layer_pair.first);        
     }
     //Create StereoAndMatchedMEs
-    std::map<std::string, StereoAndMatchedMEs>::iterator iStereoAndMatchedME  = StereoAndMatchedMEsMap.find(label);
-    if(iStereoAndMatchedME==StereoAndMatchedMEsMap.end()) {
+    if(StereoAndMatchedMEsMap.find(label)==StereoAndMatchedMEsMap.end()) {
 	
       // get detids for the stereo and matched layer. We are going to need a bool for these layers
       bool isStereo = false;
       // Keep in mind that when we are on the TID or TEC we deal with rings not wheel 
-      int32_t stereolnumber = det_layer_pair.second;
       std::vector<uint32_t> stereoandmatchedDetIds;        
-      if ( (det_layer_pair.first == "TIB") &&  (tTopo->tibIsStereo(detid) == 1) ) {
-	substructure.getTIBDetectors(activeDets,stereoandmatchedDetIds,stereolnumber,0,0,0);
-	isStereo = true;
-      } else if ( (det_layer_pair.first == "TOB") &&  (tTopo->tobIsStereo(detid) == 1) ) {
-	substructure.getTOBDetectors(activeDets,stereoandmatchedDetIds,stereolnumber,0,0);
-	isStereo = true;
-      } else if ( (det_layer_pair.first == "TID") && (stereolnumber > 0) && (tTopo->tidIsStereo(detid) == 1) ) {
-	substructure.getTIDDetectors(activeDets,stereoandmatchedDetIds,2,0,abs(stereolnumber),1);
-	isStereo = true;
-      } else if ( (det_layer_pair.first == "TID") && (stereolnumber < 0) && (tTopo->tidIsStereo(detid) == 1) ) {
-	substructure.getTIDDetectors(activeDets,stereoandmatchedDetIds,1,0,abs(stereolnumber),1);
-	isStereo = true;
-      } else if ( (det_layer_pair.first == "TEC") && (stereolnumber > 0) && (tTopo->tecIsStereo(detid) == 1) ) {
-	substructure.getTECDetectors(activeDets,stereoandmatchedDetIds,2,0,0,0,abs(stereolnumber),1);
-	isStereo = true;
-      } else if ( (det_layer_pair.first == "TEC") && (stereolnumber < 0) && (tTopo->tecIsStereo(detid) == 1) ) {
-	substructure.getTECDetectors(activeDets,stereoandmatchedDetIds,1,0,0,0,abs(stereolnumber),1);
-	isStereo = true;
+      if ( tTopo->tibIsStereo(detid) == 1 ) {
+        int32_t stereolnumber = det_layer_pair.second;
+        std::string stereolname = det_layer_pair.first;
+        if ( stereolname == "TEC" ) {
+          if ( stereolnumber > 0 ) {
+	    substructure.getTECDetectors(activeDets,stereoandmatchedDetIds,2,0,0,0,abs(stereolnumber),1);
+	    isStereo = true;
```

Comment:
If you mean to put isStereo = true outside of the case, then you get a problem with stereolnumber = 0